common/lib/xmodule/xmodule/modulestore/tests/test_publish.py

```diff
@@ -139,3 +157,1222 @@ def test_publish_draft_delete(self):
+            # Create course.
+            self.course = store.create_course(course_key.org, course_key.course, course_key.run, self.user_id)
+
+            # Create chapters.
+            block_type = 'chapter'
+            for idx in xrange(0, 2):
+                parent_type = 'course'
+                parent_id = 'course'
+                block_id = _make_block_id(block_type, idx)
+                setattr(self, block_id, _create_child(self.course, block_type, block_id))
+                db_entry = {
+                    (block_type, block_id): _make_course_db_entry(
+                        parent_type, parent_id, block_id, idx, 'sequential', 'sequential'
+                    )
+                }
+                self.course_db.update(db_entry)
+
+            # Create sequentials.
+            block_type = 'sequential'
+            for idx in xrange(0, 4):
+                parent_type = 'chapter'
+                parent_id = _make_block_id(parent_type, idx / 2)
+                parent_item = getattr(self, parent_id)
+                block_id = _make_block_id(block_type, idx)
+                setattr(self, block_id, _create_child(parent_item, block_type, block_id))
+                db_entry = {
+                    (block_type, block_id): _make_course_db_entry(
+                        parent_type, parent_id, block_id, idx, 'vertical', 'vertical'
+                    )
+                }
+                self.course_db.update(db_entry)
+
+            # Create verticals.
+            block_type = 'vertical'
+            for idx in xrange(0, 8):
+                parent_type = 'sequential'
+                parent_id = _make_block_id(parent_type, idx / 2)
+                parent_item = getattr(self, parent_id)
+                block_id = _make_block_id(block_type, idx)
+                setattr(self, block_id, _create_child(parent_item, block_type, block_id))
+                db_entry = {
+                    (block_type, block_id): _make_course_db_entry(
+                        parent_type, parent_id, block_id, idx, 'html', 'unit'
+                    )
+                }
+                self.course_db.update(db_entry)
+                self.all_verticals.append((block_type, block_id))
+
+            # Create html units.
+            block_type = 'html'
+            for idx in xrange(0, 16):
+                parent_type = 'vertical'
+                parent_id = _make_block_id(parent_type, idx / 2)
+                parent_item = getattr(self, parent_id)
+                block_id = _make_block_id('unit', idx)
+                setattr(self, block_id, _create_child(parent_item, 'html', block_id))
+                db_entry = {
+                    (block_type, block_id): _make_course_db_entry(
+                        parent_type, parent_id, block_id, idx, '', ''
+                    )
+                }
+                self.course_db.update(db_entry)
+                self.all_units.append((block_type, block_id))
```

Comment:
Would this code be easier to read/write if you used `CourseFactory` and `ItemFactory` classes, rather than the modulestore APIs directly?